File: app/src/main/java/com/example/lpiem/hearthstonecollectorapp/Models/Deck.java
package com.example.lpiem.hearthstonecollectorapp.Models;

public class Deck {
    private int id;
    private String name;
    private String description;
    private Card[] cardsList = null;

    public Deck() {
    }

    public Deck(int id, String name, String description) {
        this.id = id;
        this.name = name;
        this.description = description;
    }

    // GETTERS //

    public int getId() {
        return id;
    }

    public String getName() {
        return name;
    }

    public String getDescription() {
        return description;
    }

    public Card[] getCardsList() {
        return cardsList;
    }

    // SETTERS //

    public void setId(int id) {
        this.id = id;
    }

    public void setName(String name) {
        this.name = name;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public void setCardsList(Card[] cardsList) {
        this.cardsList = cardsList;
    }
}

File: app/src/main/java/com/example/lpiem/hearthstonecollectorapp/Models/Deck.kt
package com.example.lpiem.hearthstonecollectorapp.Models

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

class Deck(id: Int, name: String, description: String) {
    @SerializedName("id") @Expose var id: Int = 0
    @SerializedName("name") @Expose var name: String? = null
    @SerializedName("description") @Expose var description: String? = null
    @SerializedName("cards_list") @Expose var cardsList: Array<Card>? = null

    override fun toString():String {
        return "Deck{" +
                "id=" + id +
                ", name=" + name +
                ", description=" + description +
                '}'
    }

}
